app/code/Magento/GroupedProduct/Model/Quote/Item/GroupedItemQty.php

```diff
@@ -0,0 +1,91 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\GroupedProduct\Model\Quote\Item;
+
+use Magento\Framework\Model\AbstractExtensibleModel;
+use Magento\GroupedProduct\Api\Data\GroupedItemQtyInterface;
+use Magento\GroupedProduct\Api\Data\GroupedItemQtyExtensionInterface;
+
+/**
+ * Object that contains quantity information for a single associated product of a Grouped Product
+ *
+ * Class \Magento\GroupedProduct\Model\Quote\Item\GroupedItemQty
+ */
+class GroupedItemQty extends AbstractExtensibleModel implements GroupedItemQtyInterface
+{
+    /**
+     * Set associated product id
+     *
+     * @param int|string $value
+     *
+     * @return $this
+     */
+    public function setProductId($value)
```

Comment:
Setters should be removed, pass all required data via the constructor. Example [here](https://github.com/magento/inventory/blob/c90749314415076d3df316d55de165210418f009/InventoryInStorePickupSales/Model/Result.php#L15)